naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2407,12 +2452,16 @@ impl Parser {
                 return Err(Error::DirectiveAfterFirstGlobalDecl { directive_span });
             }
             (Token::Word("struct"), _) => {
+                ensure_no_diag_attrs("`struct`s", diagnostic_filters)?;
```

Comment:
Github doesn't let me comment above but: Doesn't the line `(Token::Separator(';'), _) => None,` also need `ensure_no_diag_attrs`?

It might be better to return an additional bool from the match statement so that we don't forget to think about this later.

Reply:
Ah, yep, you're right. I missed this case.

Fixed with the force-pushed f9ac43af.

Reply:
Thoughts on returning a bool?

Reply:
I'm not sure what you're suggesting, but I can imagine returning `let diag_attrs_disallowed_case: Option<&str> = match { /* … */ };`. I don't think that this is better than calling `ensure_no_diag_attrs` in each branch, though, because each branch has its own parsing code after the check. IOW, moving the check to be after the `match` would defer errors on earlier spans of the code until after parsing of, e.g., the `struct` body was done.